contracts/core/Allo.sol

```diff
@@ -124,8 +133,19 @@ contract Allo is Initializable, Ownable, AccessControl {
     /// =========== Modifier ===============
     /// ====================================
 
-    modifier onlyPoolOwner(uint256 _poolId) {
-        if (!hasRole(keccak256(abi.encodePacked(_poolId)), msg.sender)) {
+    // NOTE: We have two onlyPoolManager modifiers in the same contract, one without parameters and one with a parameter (_poolId). The other one
+    // is in the BaseStrategy.sol contract. This is not allowed in Solidity. Caught this trying to use the protocol in a demo.
+    // I appended WithPoolId to the modifier name to make it unique. Happy to rename to what the team decides.
+    modifier onlyPoolManagerWithPoolId(uint256 _poolId) virtual {
+        if (!_isPoolManager(_poolId, msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+        _;
```

Comment:
Huh? I don't see any problem with this, since they aren't in the same contract. What am I missing?

Reply:
Was trying to use our contracts in an example and had issues with duplicate names in different inherited contracts:
```javascript
/// @author @codenamejason <jax@jaxcoder.xyz>
contract StrategyRunner is Allo, TokenBalanceAllocationGating {
    /// @notice Constructor
    /// @param _owner The address of the owner of this contract.
    constructor(address _owner, address _allo) {
        // todo:
    }

    /// @notice Modifier to check if the caller is the Allo contract
    modifier onlyAllo() override {
        if (msg.sender != address(allo) && address(allo) != address(0)) {
            revert UNAUTHORIZED();
        }
        _;
    }

    modifier onlyPoolManager() override(BaseStrategy) {
        // todo:
        _;
    }

    modifier onlyPoolManagerWithPoolId(uint256 _poolId) override {
        // todo:
        _;
    }

    receive() external payable {}
}
```

Reply:
Was experimenting with some implementations...

Reply:
Going to revert this change.